sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -680,6 +681,18 @@ class QueryCompilationErrorsSuite
       context = ExpectedContext("", "", 7, 13, "CAST(1)")
     )
   }
+
+  test("NEGATIVE_SCALE_NOT_ALLOWED: negative scale for Decimal is not allowed") {
```

Comment:
Could you remove the error class name from test's title, please.

Reply:
And let's move this test to `DecimalSuite` if there is no such test there.

Reply:
Thanks! Moved  and migrate the existing test into `checkError`.